app.js

```diff
@@ -83,37 +84,78 @@ e.g.
 */
 app.use('/interactive', (req, res) => {
   const parsed = JSON.parse(req.body.payload)
-  const {
-    // type,
-    response_url,
-    // user,
-    // channel: { id, name },
-    // TODO when would this be .length > 1
-    actions: [{
-      value: action_value,
-      // style,
-      // type: action_type,
-      // text: { text },
-    }],
 
+  const {
+    type,
+    view: {
+      id,
+      hash,
+      callback_id,
+      private_metadata,
+      state: { values }, // only available from submission
+      blocks,
+    },
+    actions,
+    user: { id: sender },
+    // much more avb inside, check documentation
   } = parsed
-  // value standard: '[module] // param-1 // ...param-N'
-  const [module, taskId, customFieldId, value] = action_value.split(' // ')
-  if (module === 'vacay') {
-    updateTask(taskId, { custom_fields: { [customFieldId]: value }})
-      .then(res => {
-        const { assignee: { name }, start_on, due_on, custom_fields } = res
-        const status = custom_fields.find(o => o.name === 'Status').enum_value.name
-        const date = start_on ? `${start_on} - ${due_on}` : due_on
-        axios.post(response_url, {
-          text: `Updated *${name}'s* vacation on *${date}* to *${status}*`,
-          mrkdwn: true,
-          response_type: 'ephemeral',
-          replace_original: false,
-        })
+
+  if (callback_id === 'bday') {
+    // manipulate data received from submission
+    const { data = {}, errors = {} } = bdayInteractive({ type, values })
+    if (Object.values(errors).length) {
+      return res.status(200).json({ response_action: 'errors', errors })
+    }
+    const {
+      ref,
+      response_url,
+      command,
+      channel_id,
+    } = JSON.parse(private_metadata)
+
+    const payload = {
+      ref,
+      command,
+      type,
+      view_id: id,
+      hash,
+      response_url,
+      data,
+      blocks,
+      action: actions ? actions[0] : [], // how to check for existence & destructure & set default value?
```

Comment:
could try `actions[0] || []`

Reply:
it errors in the console because it can't iterate over undefined :(
I tried `actions?[0]` but it didn't like it either, it thinks it's a ternary expression and expects the `:`

Reply:
oh lol, does actions array only have one item? could try to assign default `[]` from when `actions` is returned like

```
const { actions = [ ] } = parsed
```